pkg/reconciler/internal/fake/actionclient.go

```diff
@@ -133,6 +140,11 @@ func (c *ActionClient) Upgrade(name, namespace string, chrt *chart.Chart, vals m
 	return c.HandleUpgrade()
 }
 
+func (c *ActionClient) MarkFailed(rel *release.Release, reason string) error {
```

Comment:
I really like the idea to mark the release as failed and being reconciled by the existing logic. Nice! 💯